src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -55,7 +55,7 @@ public function dump(mixed $input, int $inline = 0, int $indent = 0, int $flags
         }
 
         if ($inline <= 0 || (!\is_array($input) && !$input instanceof TaggedValue && $dumpObjectAsInlineMap) || !$input) {
-            $output .= $prefix.Inline::dump($input, $flags);
+            $output .= $prefix.(0 <= $inline && null === $input && $flags & Yaml::DUMP_NULL_AS_EMPTY_IN_EXPANDED_NOTATION ? '' : Inline::dump($input, $flags));
```

Comment:
The location of this code change looks weird to me. What happens when `$inline >= 0` (i.e. we haven't reached the level where we force inline values yet) ? All you test put `null` on the second level of nesting, while switching to inline mode at depth 2 or less (there is a case putting `null` on the first level, but it has the inline depth set at `1` as well).

I'm quite sure this will *not* dump null values as empty content properly in expanded maps that are less nested than the limit. And on the other hand, it looks like it could dump a top-level `null` as empty value with the default value for `$inline` (which is not covered by your tests)